flexmeasures_openweathermap/utils/locating.py

```diff
@@ -92,19 +92,35 @@ def find_weather_sensor_by_location(
         n=1,
     )
     if weather_sensor is not None:
-        weather_station: Asset = weather_sensor.generic_asset
+        weather_station: GenericAsset = weather_sensor.generic_asset
         if abs(
             location[0] - weather_station.location[0]
         ) > max_degree_difference_for_nearest_weather_sensor or abs(
             location[1] - weather_station.location[1]
             > max_degree_difference_for_nearest_weather_sensor
         ):
-            current_app.logger.warning(
+            raise Warning(
                 f"[FLEXMEASURES-OWM] No sufficiently close weather sensor found (within {max_degree_difference_for_nearest_weather_sensor} {flexmeasures_inflection.pluralize('degree', max_degree_difference_for_nearest_weather_sensor)} distance) for measuring {sensor_name}! We're looking for: {location}, closest available: ({weather_station.location})"
             )
     else:
-        current_app.logger.warning(
+        raise Warning(
             "[FLEXMEASURES-OWM] No weather sensor set up yet for measuring %s. Try the register-weather-sensor CLI task."
             % sensor_name
         )
     return weather_sensor
```

Comment:
`test_get_weather_forecasts_no_close_sensors` test was failing because of these warnings.

Reply:
tests are passing but there's a bug.

Reply:
I think we need to revisit the tests then, not change the code so the tests pass.

Reply:
```python
   if abs(
            location[0] - weather_station.location[0]
        ) > max_degree_difference_for_nearest_weather_sensor or abs(
            location[1] - weather_station.location[1]
            > max_degree_difference_for_nearest_weather_sensor
        ):
            raise Warning(
                f"[FLEXMEASURES-OWM] No sufficiently close weather sensor found (within {max_degree_difference_for_nearest_weather_sensor} {flexmeasures_inflection.pluralize('degree', max_degree_difference_for_nearest_weather_sensor)} distance) for measuring {sensor_name}! We're looking for: {location}, closest available: ({weather_station.location})"
            )

``` 

If we raise warning this way, then the program would be stopped because of no closer station to location. `current_app.logger.warning` doesn't stop the program and that's why the tests are failing.